merkledag/merkledag_test.go

```diff
@@ -129,26 +126,76 @@ func TestBatchFetchDupBlock(t *testing.T) {
 	runBatchFetchTest(t, read)
 }
 
+// makeTestDAG creates a simple DAG from the data in a reader.
+// First, a node is created from each 512 bytes of data from the reader
+// (like a the Size chunker would do). Then all nodes are added as children
+// to a root node, which is returned.
+func makeTestDAG(t *testing.T, read io.Reader, ds ipld.DAGService) ipld.Node {
+	p := make([]byte, 512)
+	nodes := []*ProtoNode{}
+	var err error
+	n, err = io.ReadFull(read, p)
+	if n != len(p) {
+		t.Fatal("should have read 512 bytes from the reader")
+	}
+	for err == nil {
+		protoNode := NodeWithData(p)
+		nodes = append(nodes, protoNode)
+		_, err = read.Read(p)
```

Comment:
Same. Needs a ReadFull. I'd just deduplicate the code and have a `for {` loop with a break/return.